src/ai/backend/manager/clients/agent/client.py

```diff
@@ -177,45 +191,54 @@ async def restart_kernel(
                 kernel_id,
                 image_ref,
                 update_config,
+                agent_id=agent_id,
             )
 
     @agent_client_resilience.apply()
-    async def sync_kernel_registry(self, kernel_tuples: list[tuple[str, str]]) -> None:
+    async def sync_kernel_registry(
+        self,
+        kernel_tuples: list[tuple[str, str]],
+        agent_id: AgentId,
+    ) -> None:
         """Sync kernel registry on the agent."""
         async with self._with_connection() as rpc:
-            return await rpc.call.sync_kernel_registry(kernel_tuples)
+            return await rpc.call.sync_kernel_registry(kernel_tuples, agent_id=agent_id)
 
     @agent_client_resilience.apply()
-    async def drop_kernel_registry(self, kernel_id_list: list[KernelId]) -> None:
+    async def drop_kernel_registry(self, kernel_id_list: list[KernelId], agent_id: AgentId) -> None:
         """Drop kernel registry entries on the agent."""
         async with self._with_connection() as rpc:
-            await rpc.call.drop_kernel_registry(kernel_id_list)
+            await rpc.call.drop_kernel_registry(kernel_id_list, agent_id=agent_id)
```

Comment:
I think it would be good if we didn't receive the agent_id every time in the agent client's method, but instead put it as a field in the agent client and have the agent client fill it. (Because I don't want to change the arguments of the method)